SIPS/sip-83.md

```diff
@@ -47,55 +71,111 @@ allows subsequent minting and burning transactions to rely on the calculated deb
 
 <!--This is a high level overview of *how* the SIP will solve the problem. The overview should clearly describe how the new feature will be implemented.-->
 
-To solve the problem of requiring every minting, burning and claiming rewards transaction to read 40+ synth prices from external oracles, the debt pool snapshot will allow the total system's issued synths value to be re-used for calculating how much debt and the relative % shift someone is issuing / burning based on the debt pool snapshot.
+#### System Debt Caching
 
-The debt pool snapshot captures a snapshot of all the synths prices (excluding sUSD) at the time and mitigates the need to retrieve latest synth prices everytime, relying on the fact that Synthetix exchanges involve repricing one synths into another based on their respective USD prices.
+To solve the problem of requiring every minting, burning and claiming rewards transaction to read 40+ synth prices from
+external oracles, the debt pool snapshot will allow the total system's issued synths value to be re-used for calculating
+how much debt and the relative % shift someone is issuing / burning based on the debt pool snapshot.
+As a system-wide value, the snapshot will be stored in the flexible storage contract described in [SIP 64](sip-64.md).
 
-The projected cost of reading all 40+ synth prices on-chain from external oracles to calculate the debt pool on each transaction is `~800,000 gas` providing a huge reduction for stakers on Synthetix to issue, burn and claim rewards.
+The debt pool snapshot captures the current synth prices and supplies at the time of the snapshot and removes the need
+to retrieve latest data on every operation, relying on the fact that Synthetix exchanges involve repricing one synths
+into another based on their respective USD prices.
 
-The cumulative debt delta allows someone issuing and burning debt to update their % ratio of the debt pool afterwards based on the latest total debt pool snapshot. During a minting and burning transaction, the debt pool snapshot will be updated with the amount of sUSD debt that is issued or burned as the issuance / burning of sUSD debt will change the debt pool size but not require latest prices to be retrieved.
+We propose a public function that can be called to update the debt pool snapshot and the caller will be paying the
```

Comment:
To solve for the aforementioned issue wrt invalid prices, as well as persisting the totalIssuedSynths, we also need to persist the invalid flag.

Reply:
@justinjmoses  good point on invalid rates and capturing them as a single flag would be optimal.

Reply:
Addressed in the Liveness section of the SIP.